src/dex-ui/hooks/useSwapData.tsx

```diff
@@ -10,56 +11,64 @@ import { useDexContext } from ".";
  * which swap data will be refetched and refreshed in the UI. If the refresh
  * interval is set to 0, swap data will not refresh on a timed interval.
  */
-export const useSwapData = (selectedRoute: any, refreshInterval = 0) => {
+export const useSwapData = (selectedPairContractId: string, refreshInterval = 0) => {
   const isInitialFetch = useRef(true);
+  const refreshTimeInterval = useRef<NodeJS.Timer>();
   const { wallet, swap } = useDexContext(({ wallet, swap }) => ({
     wallet,
     swap,
   }));
   const walletAccountId = wallet.savedPairingData?.accountIds[0];
   const { transactionState, fetchFee, fetchSpotPrices, getPrecision, fetchTokenPairs } = swap;
 
+  async function fetchSwapDataOnLoad() {
+    await Promise.allSettled([
+      fetchFee(selectedPairContractId),
+      fetchSpotPrices(selectedPairContractId),
+      fetchTokenPairs(),
+      getPrecision(selectedPairContractId),
+    ]);
+  }
+
   /**
    * Fetches all swap data on first load and change of wallet account ID.
    */
   useEffect(() => {
-    async function fetchSwapDataOnLoad() {
-      await Promise.allSettled([
-        fetchFee(selectedRoute.selectedPairId),
-        fetchSpotPrices(selectedRoute.selectedPairId, selectedRoute.selectedAToBRoute, selectedRoute.selectedBToARoute),
-        fetchTokenPairs(),
-        getPrecision(selectedRoute.selectedPairId),
-      ]);
-    }
     fetchSwapDataOnLoad();
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [walletAccountId]);
 
+  async function fetchSwapDataOnInterval() {
+    await fetchSpotPrices(selectedPairContractId);
+  }
   /**
    * Fetches all swap data on a specified interval (milliseconds).
    */
   useEffect(() => {
```

Comment:
Fixed swap data interval logic to only fetch swap data when a pair of tokens are selected in the UI.